#!/bin/bash 
#SBATCH -D /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs
#SBATCH -J GATK_DBI_genotyping_filtering_mito
#SBATCH --account=co_rosalind
#SBATCH --partition=savio
#SBATCH --qos=rosalind_savio_normal
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=20
#SBATCH --time=400:00:00
#SBATCH --mem=64000
#SBATCH -o /global/scratch2/peter_stokes/err_outs/GATK_DBI_genotyping_filtering_mito.out
#SBATCH -e /global/scratch2/peter_stokes/err_outs/GATK_DBI_genotyping_filtering_mito.err
#SBATCH --mail-user=peter_stokes@berkeley.edu
#SBATCH --mail-type=All

module load java
module load bio/vcftools/0.1.15
module load htslib/1.6

/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx128g" GenomicsDBImport \
--genomicsdb-workspace-path /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/DBI_mito \
--reader-threads 12 \
--batch-size 75 \
-L HanXRQMT \
-R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta \
--variant ANN0808.g.vcf.gz \
--variant ANN0809.g.vcf.gz \
--variant ANN0810.g.vcf.gz \
--variant ANN0814.g.vcf.gz \
--variant ANN0815.g.vcf.gz \
--variant ANN0820.g.vcf.gz \
--variant ANN0828.g.vcf.gz \
--variant ANN0829.g.vcf.gz \
--variant ANN0830.g.vcf.gz \
--variant ANN0833.g.vcf.gz \
--variant ANN0835.g.vcf.gz \
--variant ANN0836.g.vcf.gz \
--variant ANN0847.g.vcf.gz \
--variant ANN0849.g.vcf.gz \
--variant ANN0850.g.vcf.gz \
--variant ANN0852.g.vcf.gz \
--variant ANN0854.g.vcf.gz \
--variant ANN0860.g.vcf.gz \
--variant ANN0863.g.vcf.gz \
--variant ANN0867.g.vcf.gz \
--variant ANN0868.g.vcf.gz \
--variant ANN0874.g.vcf.gz \
--variant ANN0876.g.vcf.gz \
--variant ANN0880.g.vcf.gz \
--variant ANN0882.g.vcf.gz \
--variant ANN0887.g.vcf.gz \
--variant ANN0889.g.vcf.gz \
--variant ANN0891.g.vcf.gz \
--variant ANN0892.g.vcf.gz \
--variant ANN0898.g.vcf.gz \
--variant ANN0905.g.vcf.gz \
--variant ANN0906.g.vcf.gz \
--variant ANN0908.g.vcf.gz \
--variant ANN0917.g.vcf.gz \
--variant ANN0918.g.vcf.gz \
--variant ANN0920.g.vcf.gz \
--variant ANN0926.g.vcf.gz \
--variant ANN0927.g.vcf.gz \
--variant ANN0928.g.vcf.gz \
--variant ANN0931.g.vcf.gz \
--variant ANN0932.g.vcf.gz \
--variant ANN0933.g.vcf.gz \
--variant ANN0945.g.vcf.gz \
--variant ANN0946.g.vcf.gz \
--variant ANN0947.g.vcf.gz \
--variant ANN0956.g.vcf.gz \
--variant ANN0957.g.vcf.gz \
--variant ANN0959.g.vcf.gz \
--variant ANN0962.g.vcf.gz \
--variant ANN0963.g.vcf.gz \
--variant ANN0964.g.vcf.gz \
--variant ANN0972.g.vcf.gz \
--variant ANN0975.g.vcf.gz \
--variant ANN0978.g.vcf.gz \
--variant ANN0981.g.vcf.gz \
--variant ANN0985.g.vcf.gz \
--variant ANN0986.g.vcf.gz \
--variant ANN0992.g.vcf.gz \
--variant ANN0996.g.vcf.gz \
--variant ANN0997.g.vcf.gz \
--variant ANN1002.g.vcf.gz \
--variant ANN1003.g.vcf.gz \
--variant ANN1004.g.vcf.gz \
--variant ANN1011.g.vcf.gz \
--variant ANN1012.g.vcf.gz \
--variant ANN1019.g.vcf.gz \
--variant ANN1025.g.vcf.gz \
--variant ANN1026.g.vcf.gz \
--variant ANN1029.g.vcf.gz \
--variant ANN1032.g.vcf.gz \
--variant ANN1034.g.vcf.gz \
--variant ANN1040.g.vcf.gz \
--variant ANN1043.g.vcf.gz \
--variant ANN1050.g.vcf.gz \
--variant ANN1053.g.vcf.gz \
--variant ANN1058.g.vcf.gz \
--variant ANN1060.g.vcf.gz \
--variant ANN1061.g.vcf.gz \
--variant ANN1062.g.vcf.gz \
--variant ANN1065.g.vcf.gz \
--variant ANN1073.g.vcf.gz \
--variant ANN1074.g.vcf.gz \
--variant ANN1080.g.vcf.gz \
--variant ANN1083.g.vcf.gz \
--variant ANN1084.g.vcf.gz \
--variant ANN1088.g.vcf.gz \
--variant ANN1092.g.vcf.gz \
--variant ANN1095.g.vcf.gz \
--variant ANN1096.g.vcf.gz \
--variant ANN1106.g.vcf.gz \
--variant ANN1107.g.vcf.gz \
--variant ANN1109.g.vcf.gz \
--variant ANN1115.g.vcf.gz \
--variant ANN1118.g.vcf.gz \
--variant ANN1120.g.vcf.gz \
--variant ANN1123.g.vcf.gz \
--variant ANN1124.g.vcf.gz \
--variant ANN1127.g.vcf.gz \
--variant ANN1132.g.vcf.gz \
--variant ANN1133.g.vcf.gz \
--variant ANN1134.g.vcf.gz \
--variant ANN1141.g.vcf.gz \
--variant ANN1143.g.vcf.gz \
--variant ANN1144.g.vcf.gz \
--variant ANN1152.g.vcf.gz \
--variant ANN1154.g.vcf.gz \
--variant ANN1160.g.vcf.gz \
--variant ANN1162.g.vcf.gz \
--variant ANN1164.g.vcf.gz \
--variant ANN1169.g.vcf.gz \
--variant ANN1172.g.vcf.gz \
--variant ANN1174.g.vcf.gz \
--variant ANN1178.g.vcf.gz \
--variant ANN1183.g.vcf.gz \
--variant ANN1187.g.vcf.gz \
--variant ANN1188.g.vcf.gz \
--variant ANN1195.g.vcf.gz \
--variant ANN1197.g.vcf.gz \
--variant ANN1198.g.vcf.gz \
--variant ANN1202.g.vcf.gz \
--variant ANN1203.g.vcf.gz \
--variant ANN1204.g.vcf.gz \
--variant ANN1216.g.vcf.gz \
--variant ANN1218.g.vcf.gz \
--variant ANN1219.g.vcf.gz \
--variant ANN1251.g.vcf.gz \
--variant ANN1252.g.vcf.gz \
--variant ANN1260.g.vcf.gz \
--variant ANN1263.g.vcf.gz \
--variant ANN1264.g.vcf.gz \
--variant ANN1268.g.vcf.gz \
--variant ANN1273.g.vcf.gz \
--variant ANN1274.g.vcf.gz \
--variant ANN1278.g.vcf.gz \
--variant ANN1284.g.vcf.gz \
--variant ANN1285.g.vcf.gz \
--variant ANN1286.g.vcf.gz \
--variant ANN1296.g.vcf.gz \
--variant ANN1297.g.vcf.gz \
--variant ANN1300.g.vcf.gz \
--variant ANN1301.g.vcf.gz \
--variant ANN1309.g.vcf.gz \
--variant ANN1310.g.vcf.gz \
--variant ANN1311.g.vcf.gz \
--variant ANN1316.g.vcf.gz \
--variant ANN1319.g.vcf.gz \
--variant ANN1322.g.vcf.gz \
--variant ANN1328.g.vcf.gz \
--variant ANN1330.g.vcf.gz \
--variant ARG0010.g.vcf.gz \
--variant ARG0054.g.vcf.gz \
--variant ARG0204.g.vcf.gz \
--variant ARG0169.g.vcf.gz \
--variant ARG0251.g.vcf.gz \
--variant PET0316.g.vcf.gz \
--variant PET0337.g.vcf.gz \
--variant PET0372.g.vcf.gz \
--variant PET0428.g.vcf.gz \
--variant PET0636.g.vcf.gz \
--variant PET0461.g.vcf.gz \
--variant PET0473N.g.vcf.gz \
--variant PET0485.g.vcf.gz \
--variant PET0550.g.vcf.gz \
--variant PET0580.g.vcf.gz \
--variant PET0654.g.vcf.gz \
--variant PET0669.g.vcf.gz \
--variant PET0680.g.vcf.gz \
--variant PET0712.g.vcf.gz \
--variant PET0732.g.vcf.gz \
--variant Ann1238_subset.g.vcf.gz \
--variant Anzac_Pueblo.g.vcf.gz \
--variant Ari_Man.g.vcf.gz \
--variant Ari_broad.g.vcf.gz \
--variant Arikara.g.vcf.gz \
--variant Arikara_122976.g.vcf.gz \
--variant Arikara_126306.g.vcf.gz \
--variant Arikara_14042_874.g.vcf.gz \
--variant Havasupai.g.vcf.gz \
--variant Hidatsa.g.vcf.gz \
--variant Hopi_dye_NEW.g.vcf.gz \
--variant Hopi_mapping_NEW.g.vcf.gz \
--variant Hopi_real_NEW.g.vcf.gz \
--variant IA1W-1.g.vcf.gz \
--variant IA1W-11R.g.vcf.gz \
--variant IA1W-2.g.vcf.gz \
--variant IA1W-5.g.vcf.gz \
--variant IA2W-11.g.vcf.gz \
--variant IA2W-17.g.vcf.gz \
--variant IA2W-38.g.vcf.gz \
--variant IA2W-7.g.vcf.gz \
--variant KS1W-2.g.vcf.gz \
--variant KS1W-27.g.vcf.gz \
--variant KS1W-32.g.vcf.gz \
--variant KS1W-35.g.vcf.gz \
--variant KS2W-15.g.vcf.gz \
--variant KS2W-24.g.vcf.gz \
--variant KS2W-32.g.vcf.gz \
--variant KS2W-35.g.vcf.gz \
--variant Laguna_Pueblo.g.vcf.gz \
--variant MB1W-16.g.vcf.gz \
--variant MB1W-16B.g.vcf.gz \
--variant MB1W-39.g.vcf.gz \
--variant MB1W-40.g.vcf.gz \
--variant MB1W-8.g.vcf.gz \
--variant MO1W-3.g.vcf.gz \
--variant MO1W-33.g.vcf.gz \
--variant MO1W-38.g.vcf.gz \
--variant Mandan1_PPN046.g.vcf.gz \
--variant Mandan2.g.vcf.gz \
--variant MexCult1.g.vcf.gz \
--variant MexCult15.g.vcf.gz \
--variant MexCult3.g.vcf.gz \
--variant MexCult6.g.vcf.gz \
--variant MexCult9.g.vcf.gz \
--variant ND1W-31.g.vcf.gz \
--variant ND1W-38.g.vcf.gz \
--variant Paiute.g.vcf.gz \
--variant SD1W-13.g.vcf.gz \
--variant SD1W-15.g.vcf.gz \
--variant SD1W-35.g.vcf.gz \
--variant SD1W-4.g.vcf.gz \
--variant SD2W-18.g.vcf.gz \
--variant SD2W-23.g.vcf.gz \
--variant SD2W-23R.g.vcf.gz \
--variant SD2W-27.g.vcf.gz \
--variant SK1W-22R.g.vcf.gz \
--variant SK1W-29R.g.vcf.gz \
--variant SK1W-37.g.vcf.gz \
--variant SK1W-Q.g.vcf.gz \
--variant San_Ildefonso_Pueblo.g.vcf.gz \
--variant Se_Purp.g.vcf.gz \
--variant Se_Strip_682a.g.vcf.gz \
--variant Se_Strip_682b.g.vcf.gz \
--variant Seneca.g.vcf.gz \
--variant Seneca_137749.g.vcf.gz \
--variant Zuni.g.vcf.gz \
--variant ann01_cwIA.g.vcf.gz \
--variant ann04_nwAR.g.vcf.gz \
--variant ann05_ccNM.g.vcf.gz \
--variant ann10_ccIL.g.vcf.gz \
--variant ann20_seAZ.g.vcf.gz \
--variant ann28_swSK.g.vcf.gz \
--variant ann29_neSD.g.vcf.gz \
--variant ann34_cwKS.g.vcf.gz \
--variant ann39_ncNE.g.vcf.gz \
--variant annMex_A1516.g.vcf.gz \
--variant annMex_A1572.g.vcf.gz \
--variant annMex_Ann261.g.vcf.gz \
--variant Coahuila_28343.g.vcf.gz \
--variant Guanajuato_29096.g.vcf.gz \
--variant Nuevo_Leon_28338.g.vcf.gz \
--variant San_Luis_Potosi_28315.g.vcf.gz \
--variant Sonora_28388.g.vcf.gz \
--variant CuevadelGallo.g.vcf.gz \
--variant GreenBluff.g.vcf.gz \
--variant SF_01.g.vcf.gz \
--variant SF_02.g.vcf.gz \
--variant SF_03.g.vcf.gz \
--variant SF_04.g.vcf.gz \
--variant SF_05.g.vcf.gz \
--variant SF_06.g.vcf.gz \
--variant SF_07.g.vcf.gz \
--variant SF_09_a.g.vcf.gz \
--variant SF_09_b.g.vcf.gz \
--variant SF_10.g.vcf.gz \
--variant SF_11_a.g.vcf.gz \
--variant SF_11_b.g.vcf.gz \
--variant SF_11_c.g.vcf.gz \
--variant SF_12.g.vcf.gz \
--variant SF_13.g.vcf.gz \
--variant SF_14.g.vcf.gz \
--variant SF_15.g.vcf.gz \
--variant SF_17.g.vcf.gz \
--variant SF_18.g.vcf.gz \
--variant SF_19.g.vcf.gz \
--variant SF_20.g.vcf.gz \
--variant SF_21.g.vcf.gz \
--variant SF_22.g.vcf.gz \
--variant SF_23.g.vcf.gz \
--variant SF_25.g.vcf.gz \
--variant SF_26.g.vcf.gz \
--variant SF_27.g.vcf.gz \
--variant SF_28.g.vcf.gz \
--variant SF_29.g.vcf.gz \
--variant SF_30.g.vcf.gz \
--variant SF_31.g.vcf.gz \
--variant SF_32.g.vcf.gz \
--variant SF_33.g.vcf.gz \
--variant SF_35.g.vcf.gz \
--variant SF_35b.g.vcf.gz \
--variant SF_36.g.vcf.gz \
--variant SF_36b.g.vcf.gz \
--variant SF_36c.g.vcf.gz \
--variant SF_36d.g.vcf.gz \
--variant SF_37.g.vcf.gz \
--variant SF_37b.g.vcf.gz \
--variant SF_38.g.vcf.gz \
--variant SF_38b.g.vcf.gz \
--variant SF_39.g.vcf.gz \
--variant SF_40.g.vcf.gz \
--variant SF_41.g.vcf.gz \
--variant SF_42.g.vcf.gz \
--variant SF_43.g.vcf.gz \
--variant SF_45.g.vcf.gz \
--variant SF_46.g.vcf.gz \
--variant SF_47.g.vcf.gz \
--variant SF_48.g.vcf.gz \
--variant SF_49.g.vcf.gz \
--variant SF_50.g.vcf.gz \
--variant SF_51.g.vcf.gz \
--variant SF_52.g.vcf.gz \
--variant SF_53.g.vcf.gz \
--variant SF_55.g.vcf.gz \
--variant SF_56.g.vcf.gz \
--variant SF_57.g.vcf.gz \
--variant SF_58.g.vcf.gz \
--variant SF_59.g.vcf.gz \
--variant SF_60.g.vcf.gz \
--variant SF_61_a.g.vcf.gz \
--variant SF_61_b.g.vcf.gz \
--variant SF_62_a.g.vcf.gz \
--variant SF_62_b.g.vcf.gz \
--variant SF_63a.g.vcf.gz \
--variant SF_63b.g.vcf.gz \
--variant SF_64a.g.vcf.gz \
--variant SF_64b.g.vcf.gz \
--variant SF_65a_coat.g.vcf.gz \
--variant SF_65a_endo.g.vcf.gz \
--variant SF_65c_coat.g.vcf.gz \
--variant SF_65c_endo.g.vcf.gz \
--variant SF_65d.g.vcf.gz \
--variant SF_65g_coat.g.vcf.gz \
--variant SF_65g_endo.g.vcf.gz \
--variant SF_65h_coat.g.vcf.gz \
--variant SF_65h_endo.g.vcf.gz \
--variant SF_65i.g.vcf.gz \
--variant SF_65j_coat.g.vcf.gz \
--variant SF_65j_endo.g.vcf.gz \
--variant SF_65k_coat.g.vcf.gz \
--variant SF_65k_endo.g.vcf.gz \
--variant SF_65l.g.vcf.gz \
--variant SF_65m.g.vcf.gz \
--variant SF_66a.g.vcf.gz \
--variant Arg01.g.vcf.gz \
--variant Arg03.g.vcf.gz \
--variant Arg04.g.vcf.gz \
--variant Arg05.g.vcf.gz \
--variant Arg06.g.vcf.gz \
--variant Arg08.g.vcf.gz \
--variant PetPet03.g.vcf.gz \
--variant PetPet04.g.vcf.gz \
--variant PetPet07.g.vcf.gz \
--variant PetPet08.g.vcf.gz \
--variant PetPet09.g.vcf.gz \
--variant PetPet11.g.vcf.gz 

/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx128g" GenotypeGVCFs \
-R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta \
-V gendb:////global/scratch2/peter_stokes/plastome_mito_Alignment/mito/DBI_mito \
-L HanXRQMT \
-O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/vcf/GATK_mito_raw_1.vcf

/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx128g" VariantFiltration \
-R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta \
-O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/vcf/GATK_mito_filtered_mapQ30_2.vcf \
--variant /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/vcf/GATK_mito_raw_1.vcf \
--filter-expression "QD < 2.0" \
--filter-name "GATK_recommened_QD" \
--filter-expression "MQ < 30.0" \
--filter-name "GATK_30_MQ" \
--filter-expression "FS > 60.0" \
--filter-name "GATK_recommened_FS" \
--filter-expression "SOR > 3.0" \
--filter-name "GATK_recommened_SOR"
   
srun vcftools --vcf /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/vcf/GATK_mito_filtered_mapQ30_2.vcf --recode --remove-indels --remove-filtered-all --minQ 1000.0 --out /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/vcf/GATK_mito_filtered_mapQ30_SNPs_3