#!/bin/bash
#SBATCH -D /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs
#SBATCH -J GATK_mito_haploCaller_01
#SBATCH --account=co_rosalind
#SBATCH --partition=savio2_htc
#SBATCH --qos=rosalind_htc2_normal
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=6
#SBATCH --time=400:00:00
#SBATCH --mem=64000
#SBATCH -o /global/scratch2/peter_stokes/err_outs/GATK_mito_haploCaller_01.out
#SBATCH -e /global/scratch2/peter_stokes/err_outs/GATK_mito_haploCaller_01.err
#SBATCH --mail-user=peter_stokes@berkeley.edu
#SBATCH --mail-type=All

module load java

/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0808.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0808.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0809.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0809.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0810.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0810.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0814.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0814.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0815.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0815.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0820.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0820.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0828.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0828.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0829.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0829.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0830.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0830.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0833.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0833.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0835.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0835.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0836.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0836.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0847.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0847.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0849.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0849.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0850.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0850.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0852.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0852.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0854.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0854.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0860.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0860.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0863.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0863.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0867.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0867.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0868.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0868.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0874.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0874.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0876.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0876.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0880.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0880.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0882.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0882.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0887.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0887.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0889.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0889.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0891.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0891.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0892.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0892.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0898.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0898.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0905.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0905.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0906.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0906.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0908.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0908.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0917.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0917.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0918.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0918.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0920.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0920.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0926.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0926.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0927.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0927.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0928.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0928.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0931.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0931.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0932.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0932.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0933.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0933.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0945.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0945.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0946.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0946.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0947.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0947.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0956.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0956.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0957.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0957.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0959.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0959.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0962.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0962.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0963.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0963.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0964.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0964.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0972.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0972.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0975.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0975.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0978.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0978.g.vcf.gz --ERC GVCF
/global/scratch2/peter_stokes/software/gatk-4.1.7.0/gatk --java-options "-Djava.io.tmpdir=/global/scratch2/peter_stokes/temp_files/ -Xmx64G" HaplotypeCaller -R /global/scratch2/peter_stokes/references/HanXRQr2.0-SUNRISE-2.1.genome.MT.fasta -I /global/scratch2/peter_stokes/bams_mito/ANN0981.rmdup.normal.bam -ploidy 1 -L HanXRQMT -O /global/scratch2/peter_stokes/plastome_mito_Alignment/mito/gvcfs/ANN0981.g.vcf.gz --ERC GVCF
